include/aws/cryptosdk/private/session.h

```diff
@@ -103,7 +103,7 @@ AWS_CRYPTOSDK_STATIC_INLINE bool aws_cryptosdk_session_is_valid(const struct aws
     bool cmm_valid            = aws_cryptosdk_cmm_base_is_valid(session->cmm);
     bool hdr_valid            = aws_cryptosdk_hdr_is_valid(&session->header);
     bool keyring_trace_valid  = aws_cryptosdk_keyring_trace_is_valid(&session->keyring_trace);
-    bool alg_props_valid      = aws_cryptosdk_alg_properties_is_valid(session->alg_props);
+    bool alg_props_valid      = session->alg_props == NULL || aws_cryptosdk_alg_properties_is_valid(session->alg_props);
```

Comment:
I'd add parenthesis.
```c
bool alg_props_valid      = (session->alg_props == NULL || aws_cryptosdk_alg_properties_is_valid(session->alg_props));
```
And makes sure @alex-chew agree that a valid `aws_cryptosdk_session` can have a NULL `alg_props`.

Reply:
Indeed `session->alg_props` can be NULL.